rocAL/source/augmentations/geometry_augmentations/node_slice.cpp

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "node_slice.h"
+
+#include <vx_ext_rpp.h>
+
+#include "exception.h"
+
+SliceNode::SliceNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) : Node(inputs, outputs) {}
+
+void SliceNode::create_node() {
+    if (_node)
+        return;
+
+    auto max_shape = _outputs[0]->info().max_shape();
+    _slice_roi.resize(_batch_size, std::vector<uint32_t>(max_shape.size()));
+    for (uint i = 0; i < _batch_size; i++)
+        for (uint j = 0; j < max_shape.size(); j++)
+            _slice_roi[i][j] = max_shape[j];
+    const int buffer_size = _batch_size;
```

Comment:
can _batch_size be directly used here? instead of storing again in another variable?

Reply:
Is this addressed?